server/src/main/java/org/cloudfoundry/identity/uaa/scim/beans/ScimSecurityConfiguration.java

```diff
@@ -0,0 +1,198 @@
+package org.cloudfoundry.identity.uaa.scim.beans;
+
+import org.cloudfoundry.identity.uaa.oauth.UaaTokenServices;
+import org.cloudfoundry.identity.uaa.oauth.provider.authentication.OAuth2AuthenticationManager;
+import org.cloudfoundry.identity.uaa.oauth.provider.authentication.OAuth2AuthenticationProcessingFilter;
+import org.cloudfoundry.identity.uaa.oauth.provider.error.OAuth2AccessDeniedHandler;
+import org.cloudfoundry.identity.uaa.oauth.provider.error.OAuth2AuthenticationEntryPoint;
+import org.cloudfoundry.identity.uaa.security.IsSelfCheck;
+import org.cloudfoundry.identity.uaa.security.SelfCheckAuthorizationManager;
+import org.cloudfoundry.identity.uaa.security.web.CookieBasedCsrfTokenRepository;
+import org.cloudfoundry.identity.uaa.web.FilterChainOrder;
+import org.cloudfoundry.identity.uaa.web.UaaFilterChain;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.access.vote.UnanimousBased;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AnonymousConfigurer;
+import org.springframework.security.config.annotation.web.configurers.CsrfConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import static org.cloudfoundry.identity.uaa.web.AuthorizationManagersUtils.anyOf;
+
+@Configuration
+@EnableWebSecurity
+class ScimSecurityConfiguration {
+
+    @Autowired
+    @Qualifier("tokenServices")
+    private UaaTokenServices tokenServices;
+
+    @Autowired()
+    @Qualifier("oauthAuthenticationEntryPoint")
+    OAuth2AuthenticationEntryPoint oauthAuthenticationEntryPoint;
+
+    @Autowired
+    @Qualifier("emptyAuthenticationManager")
+    AuthenticationManager emptyAuthenticationManager;
+
+    @Autowired
+    @Qualifier("oauthAccessDeniedHandler")
+    OAuth2AccessDeniedHandler oauthAccessDeniedHandler;
+
+    @Autowired
+    CookieBasedCsrfTokenRepository csrfTokenRepository;
+
+    @Bean
+    @Order(FilterChainOrder.SCIM)
```

Comment:
The filter chains should not have the same order in general

**[issue]**  In particular here, `scimUserPassword` and `scimUsers` conflict ; if for some reason `scimUsers` gets picked up first, it will swallow all calls to `/Users/...`

Reply:
Good point. I do need all chains to have a order, so it ends up before the swallow all UI. In between most of them, the order doesn't mean anything with the exception that you mention. 

I will fix this.